drift/info_parser.py

```diff
@@ -140,56 +155,27 @@ def _group_comparisons(comparisons):
 
     # default values for method run
     grouped_comparisons = []
+    grouped_comparisons_with_dotnotation = {}
     processed_group_names = set()
 
-    def _get_group_name(name):
-        n, _, _ = name.partition(".")
-        return n
-
-    def _get_value_name(name):
-        _, _, n = name.partition(".")
-        return n
-
-    def _find_group(name, grouped_comps):
-        for group in grouped_comps:
-            if group["name"] == name:
-                return group
-
-    def _set_summary_state(grouped_comparison):
-        states = {comparison["state"] for comparison in grouped_comparison["comparisons"]}
-        if COMPARISON_DIFFERENT in states:
-            grouped_comparison["state"] = COMPARISON_DIFFERENT
-        elif COMPARISON_INCOMPLETE_DATA in states:
-            if COMPARISON_SAME in states:
-                grouped_comparison["state"] = COMPARISON_DIFFERENT
-            else:
-                grouped_comparison["state"] = COMPARISON_INCOMPLETE_DATA
-        elif COMPARISON_SAME in states and len(states) == 1:
-            grouped_comparison["state"] = COMPARISON_SAME
-        else:  # use 'incomplete data' as the fallback state if something goes wrong
-            grouped_comparison["state"] = COMPARISON_INCOMPLETE_DATA
-
-    # with PT_GC_BUILD_GROUP_NAMES.time(): # move to proper place or remove competely
-    # with PT_GC_POPULATE_GROUPS.time(): # move to proper place or remove competely
-    # with PT_GC_SET_SUMMARY.time(): # move to proper place or remove competely
     for comparison in comparisons:
         if "." in comparison["name"]:
-            group_name = _get_group_name(comparison["name"])
+            group_name, _, comparison_value_name = comparison["name"].partition(".")
+
             built_group = {"name": group_name, "comparisons": []}
```

Comment:
Does this need to be created each time for a grouped comparisons?

Reply:
Yes, since the group name and value can be changed. Our input is not sorted, so, we need to assume different values from the `comparison` for each item.

Reply:
Please check where the `built_group` is referenced.

```
            built_group = {"name": group_name, "comparisons": []}

            # build out group names
            if group_name not in processed_group_names:
                processed_group_names.add(group_name)
                grouped_comparisons_with_dotnotation.update({group_name: built_group})
```
Can it be move do the `if` block?

Reply:
Fixed here: c9557a969b10fd3073dda1c651bb7d573867aef5